app/src/main/java/com/spendster/presentation/addExpenses/chooseCategory/ChooseCategoryActivity.java

```diff
@@ -0,0 +1,65 @@
+package com.spendster.presentation.addExpenses.chooseCategory;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.Toast;
+
+import com.google.gson.Gson;
+import com.spendster.R;
+import com.spendster.data.entity.Category;
+
+import java.util.List;
+
+public class ChooseCategoryActivity extends AppCompatActivity implements CategoryView {
+
+    private RecyclerView recyclerView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_choose_category);
+        initUI();
+    }
+
+    private void initUI(){
+        recyclerView = findViewById(R.id.recyclerView);
+        ChooseCategoryPresenter chooseCategoryPresenter = new ChooseCategoryPresenter(this,
```

Comment:
@SimonMaha init of presenter should be in onCreate method. It has nothing common with UI to be here.